File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    let answer = 1;
    const map = new Map();
    
    clothes.forEach(item => {
        map.set(item[1], (map.get(item[1]) || 1) + 1);
    })
    
    for (const [_, value] of map) {
        answer *= value;
    }
    
    return answer - 1;
}

File: 프로그래머스/2/42578. 의상/의상.py
from collections import defaultdict

def solution(clothes):
    answer = 1
    _dict = defaultdict(int)
    
    for _, clothType in clothes:
        _dict[clothType] += 1
    
    for count in _dict.values():
        answer *= (count + 1)
    
    return answer - 1
